"UNAUTHORIZED for url: http://localhost:8765/bearer', 401)"

This is popping up everywhere. I thought it is related to a malfunction of the public httpbin instance, but I can finally replicate it locally with the docker-based one.

```
FAILED ../commands/tests/test_download.py::test_download_new_bearer_token_nosave
```

It is *never* happening when this test runs in isolation, but always, when all of `test_download.py` run.

--- a/datalad_next/utils/requests_auth.py
+++ b/datalad_next/utils/requests_auth.py
@@ -122,11 +122,17 @@ class DataladAuth(requests.auth.AuthBase):
             # get a realm ID for this authentication scheme
             realm = get_auth_realm(url, auth_schemes, scheme=ascheme)
             # ask for matching credentials
-            creds = self._credman.query(
-                _sortby='last-used',
-                type=ctype,
-                realm=realm,
-            )
+            creds = [
+                (name, cred) for name, cred in self._credman.query(
+                    _sortby='last-used',
+                    type=ctype,
+                    realm=realm,
+                )
+                # we can only work with complete credentials, although
+                # query() might return others. We exclude them here
+                # to be able to fall back on manual entry further down
+                if cred.get('secret')
+            ]
             if creds:
                 # we have matches, go with the last used one
                 name, cred = creds[0]

